src/normalize/equation.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Element, NodeEntry, Transforms } from 'slate'
+
+import { Equation } from '../interfaces'
+import { CnxEditor } from '../plugin'
+
+export default function normalizeEquation(editor: CnxEditor, entry: NodeEntry): boolean {
+    const [node, path] = entry
+
+    if (Equation.isEquation(node)) {
+        // Equations must not be empty
+        if (node.children.length === 0) {
+            Transforms.removeNodes(editor, { at: path })
+            return true
+        }
+
+        // Rules must only have a single child
+        if (node.children.length > 1) {
+            Transforms.splitNodes(editor, { at: [...path, node.children.length - 1] })
+            return true
+        }
+
+        // Equation can only contain elements from a predefined set and in
+        // a strict order.
+        if (!Element.isElement(node.children[0]) || !editor.isEquationContent(node.children[0])) {
```

Comment:
should it be only checked for the equation's first child?

Reply:
The previous `if` enforces that equations only have one child.